meinberlin/apps/dashboard/views.py

```diff
@@ -89,6 +91,86 @@ def get_permission_object(self):
         return self.organisation
 
 
+class ModulePublishView(ModuleDispatchMixin,
+                        SingleObjectMixin,
+                        generic.View):
+    permission_required = 'a4projects.change_project'
+    model = module_models.Module
+    slug_url_kwarg = 'module_slug'
+
+    def get_permission_object(self):
+        return self.project
+
+    def post(self, request, *args, **kwargs):
+        action = request.POST.get('action', None)
+        if action == 'publish':
+            self.publish_module()
+        elif action == 'unpublish':
+            self.unpublish_module()
+        else:
+            messages.warning(self.request, _('Invalid action'))
+
+        return HttpResponseRedirect(self.get_next())
+
+    def get_next(self):
+        if 'referrer' in self.request.POST:
+            return self.request.POST['referrer']
+        elif 'HTTP_REFERER' in self.request.META:
+            return self.request.META['HTTP_REFERER']
+
+        return reverse('a4dashboard:project-edit', kwargs={
+            'project_slug': self.project.slug
+        })
+
+    def publish_module(self):
```

Comment:
the publish and unpublish function are pretty similar, maybe they can be combined somehow

Reply:
Yeah, the function could know, which it gets?! But I think, it wouldn't be more readable, when we then add three more if-statements and only have two lines out of if-statements.

Reply:
fair enough